text/cognito-construct-library/working-backwards-readme.md

```diff
@@ -0,0 +1,300 @@
+# Amazon Cognito Construct Library
+
+Amazon Cognito provides authentication, authorization, and user management for your web and mobile apps. Your users can
+sign in directly with a user name and password, or through a third party such as Facebook, Amazon, Google or Apple.
+
+The two main components of Amazon Cognito are user pools and identity pools. User pools are user directories that
+provide sign-up and sign-in options for your app users. Identity pools enable you to grant your users access to other
+AWS services.
+
+This module is part of the [AWS Cloud Development Kit](https://github.com/aws/aws-cdk) project.
+
+## User Pools
+
+User pools allows creating and managing your own directory of users that can sign up and sign in. They enable easy
+integration with social identity providers such as Facebook, Google, Amazon, Microsoft Active Directory, etc. through
+SAML.
+
+The CDK APIs provide ways to both create user pools, via the `UserPool` constructor, or import existing ones that were
+created outside the CDK App.
+
+User pools can be imported using the `UserPool.fromUserPoolAttributes()` API. They can be imported either using the
+user pool ARN or the user pool id. For example:
+
+```ts
+const stack = new Stack(app, 'my-stack');
+
+UserPool.fromUserPoolAttributes(stack, 'my-imported-user-pool', {
+  userPoolId: 'us-east-1_123412341'
+});
+```
+
+> TODO: provider name and provider url return values?
+
+The following sections describe how user pools can be configured. **Unless explicitly stated, all of these
+configurations are available only when creating new user pools.**
+
+### Sign Up
+
+Users need to either signed up by the app's administrators or can sign themselves up. You can read more about both these
+kinds of sign up and how they work
+[here](https://docs.aws.amazon.com/cognito/latest/developerguide/signing-up-users-in-your-app.html).
+
+Further, a welcome email and/or SMS can be configured to be sent automatically once a user has signed up. This welcome
+email and/or SMS will carry the temporary password for the user. The user will use this password to log in and reset
+their password. The temporary password is valid only for a limited number of days.
+
+All of these options can be configured under the `signUp` property. The pool can be configured to let users sign
+themselves up by setting the `selfSignUp` property. A welcome email template can be configured by specifying the
+`welcomeEmail` property and a similar `welcomeSms` property for the welcome SMS. The validity of the temporary password
```

Comment:
I don't know that we need this level of detail here. It mostly belongs to the reference. The README should give people references to entry points, preferably in the form of code samples. In this case I would just say something like:

You can setup sign up options through the `signUp` property. For example, the following code configures a user pool to send a sign up email with a password that expires after 5 days:

```ts
<code snippet here>
```

Reply:
switched around.